pkg/services/folder/folderimpl/folder.go

```diff
@@ -66,6 +66,28 @@ func ProvideService(
 	}
 }
 
+func (s *Service) Get(ctx context.Context, cmd *folder.GetFolderQuery) (*folder.Folder, error) {
+	if s.cfg.IsFeatureToggleEnabled(featuremgmt.FlagNestedFolders) {
+		return s.store.Get(ctx, *cmd)
```

Comment:
I think we have to add a similar check whether the user has permissions to view the folder:
https://github.com/grafana/grafana/blob/53797c8ba5e7364c2997d667dc9946340bf336fd/pkg/services/folder/folderimpl/folder.go#L130-L136

I have missed it in the initial PR